src/test/java/org/sagebionetworks/bridge/services/AccountServiceTest.java

```diff
@@ -1,167 +1,954 @@
+        account.setId(USER_ID);
+
+        // execute and verify
+        service.changePassword(account, ChannelType.PHONE, DUMMY_PASSWORD);
+        verify(mockAccountDao).updateAccount(accountCaptor.capture(), eq(null));
+
+        // Simpler than changePasswordSuccess() test as we're only verifying phone is verified
+        Account updatedAccount = accountCaptor.getValue();
+        assertNull(updatedAccount.getEmailVerified());
+        assertTrue(updatedAccount.getPhoneVerified());
+        assertEquals(updatedAccount.getStatus(), ENABLED);
+    }
+
+    @Test
+    public void changePasswordForExternalId() {
+        // Set up test account
+        Account account = Account.create();
+        account.setStudyId(TEST_STUDY_IDENTIFIER);
+        account.setId(USER_ID);
+
+        // execute and verify
+        service.changePassword(account, null, DUMMY_PASSWORD);
+        verify(mockAccountDao).updateAccount(accountCaptor.capture(), eq(null));
+
+        // Simpler than changePasswordSuccess() test as we're only verifying phone is verified
+        Account updatedAccount = accountCaptor.getValue();
+        assertNull(updatedAccount.getEmailVerified());
+        assertNull(updatedAccount.getPhoneVerified());
+        assertEquals(updatedAccount.getStatus(), ENABLED);
+    }
+
+    @Test
+    public void authenticateSuccessWithHealthCode() throws Exception {
+        mockGetAccountById(PASSWORD_SIGNIN.getAccountId(), true);
+
+        Study study = Study.create();
+
+        Account account = service.authenticate(study, PASSWORD_SIGNIN);
+        assertEquals(account.getId(), USER_ID);
+        assertEquals(account.getStudyId(), TEST_STUDY_IDENTIFIER);
+        assertEquals(account.getEmail(), EMAIL);
+        assertEquals(account.getHealthCode(), HEALTH_CODE);
+        assertEquals(account.getVersion(), 1); // version not incremented by update
+    }
+
+    // This test is just a negative test to verify that the reauth token is not being rotated...
+    // regardless of how study.reauthenticationEnabled is set, it will succeed because we don't
+    // touch the reauth token
+    @Test
+    public void authenticateSuccessNoReauthentication() throws Exception {
+        Study study = Study.create();
+        study.setReauthenticationEnabled(false);
+
+        mockGetAccountById(ACCOUNT_ID_WITH_EMAIL, true);
+
+        Account account = service.authenticate(study, PASSWORD_SIGNIN);
+        // not incremented by reauthentication
+        assertEquals(account.getVersion(), 1);
+
+        // No reauthentication token rotation occurs
+        verify(mockAccountDao, never()).updateAccount(any(), eq(null));
+        assertNull(account.getReauthToken());
+    }
+
+    @Test(expectedExceptions = EntityNotFoundException.class)
+    public void authenticateAccountNotFound() throws Exception {
+        when(mockAccountDao.getAccount(ACCOUNT_ID_WITH_EMAIL)).thenThrow(new EntityNotFoundException(Account.class));
```

Comment:
This should return an optional empty instead of throwing.

Reply:
Changed